src/main/java/teammates/common/datatransfer/questions/FeedbackRankRecipientsQuestionDetails.java

```diff
@@ -238,6 +239,76 @@ public String getQuestionAdditionalInfoHtml(int questionNumber,
                         Slots.QUESTION_ADDITIONAL_INFO, additionalInfo);
     }
 
+    /**
+     * Constructs results statistics for each student.
+     * Statistics to student is only shown when visibility setting is permissive enough
```

Comment:
I believe we shall not call is as "permissive enough". All responses available in this bundle are viewable by the student. Therefore, if we remove the first two checks, the consequence would be that the student may get a meaningful summary. Think about it in this way. Technically, the student can calculate their overall rankings manually by aggregating received responses. This PR would just calculate the stats for them and that's all. So if there is a feedback session that only shows responses to giver and recipient, the student will get overall ranking `#1` though not helpful.

Reply:
Therefore, the check can just check whether we have showResponsesTo students/student's own team etc (anything that could make the stats meaningful) despite the receiver is team or not.

Reply:
> Technically, the student can calculate their overall rankings manually by aggregating received responses.

I think you're confusing overall ranking with average ranking? (Can see this https://github.com/TEAMMATES/teammates/issues/8701#issuecomment-380072745)

If not, student cannot always calculate their overall ranking because responses can be anonymous.

Reply:
You are right. :)

This means that we need to enumerate all combinations that are permissive enough. However, I think we should not put any critical logic inside the bundle. To avoid this, checking if the recipient names are available in all responses may be enough. So we may do something like this.

```java
isRecipientAvaliableInAllResponses = true
for (res : allResponses) {
    if (!isRecipientVisible(res)) {
        isRecipientAvaliableInAllResponses = false
        break
    }
}
```

Reply:
I think above mentioned condition will not always be permissive, for example when student has to give responses to team members and responses are visible only recipient then we cannot form statistics. Also above condition misses the cases where even though recipient is hidden, we can still show statistics to user.

Reply:
Yes. Thank you for correcting me.

But your checks miss a condition where students gives feedback to other students in the course and feedbacks are visible to other student?

Do you see the pain of enumerating cases? Though I cannot think of a better way according to the inputs.

Reply:
> Is that considered as permissive enough? Your checks also make sure the recipient is visible?

My checks don't consider visibility of recipient. I think visibility shouldn't have say in showing statistics (because statistics don't reveal the recipient name in anyway so anonymity is still maintained ) just like what we do in contribution questions.

> Then, we make sure that other recipients stats are in the map

What if the recipients are instructors? What if the the recipient is self?

> Do you see the pain of enumerating cases?

So this whole logic of overall ranking is analogous to Contribution question. 
![contrib](https://user-images.githubusercontent.com/24381442/42434664-a0febdc8-8371-11e8-8450-b2f36bad7e66.png)
If we can just test that above mentioned visibility and feedbackpath settings are satisfied + the case when the response giver is team, I think there wouldn't be anymore cases to look for?

Reply:
OK. I get the idea, so you are working on a new feature that shows extra stats for students. The stats will only be shown in some certain conditions (so we don't have to enumerate all conditions).